### ISSUE TITLE
POST /locations

### ISSUE THREAD

**Issue by timoweiss** [locator-kn/cor#43]
- Image upload (multipart/formdata)
- the formdata should contain:
  - title
  - lat
  - long
  - categories
  - file

Output from @SteffenGorenflo (ms-fileserve):

```
{
  "user_id": "569e4a83a6e5bb503b838301",
  "title": "Locator Headquarter",
  "categories": [],
  "favorites": [],
  "description": "",
  "images": {

    "xlarge": "/api/v2/locations/impression/image/569e4a974c9d7b5f3b4005c3/locator_headquarter.jpeg",
    "large": "/api/v2/locations/impression/image/569e4a974c9d7b5f3b4005c7/locator_headquarter.jpeg",
    "normal": "/api/v2/locations/impression/image/569e4a974c9d7b5f3b4005c9/locator_headquarter.jpeg",
    "small": "/api/v2/locations/impression/image/569e4a974c9d7b5f3b4005cb/locator_headquarter.jpeg"
  },
  "public": true,
  "geotag": {
    "type": "Point",
    "coordinates": [
      9.169753789901733,
      47.66868204997508
    ]
  }
}
```

Output (data that lands in the db) should look like the following:

```
{
  "_id": "569e4a9a4c9d7b5f3b400709",
  "create_date": "2015-07-22T10:19:22.293Z",
  "modified_date": "2015-07-27T07:24:06.381Z",
  "user_id": "569e4a83a6e5bb503b838301",
  "title": "Locator Headquarter",
  "categories": [],
  "favorites": [],
  "description": "",
  "images": {

    "xlarge": "/api/v2/locations/impression/image/569e4a974c9d7b5f3b4005c3/locator_headquarter.jpeg",
    "large": "/api/v2/locations/impression/image/569e4a974c9d7b5f3b4005c7/locator_headquarter.jpeg",
    "normal": "/api/v2/locations/impression/image/569e4a974c9d7b5f3b4005c9/locator_headquarter.jpeg",
    "small": "/api/v2/locations/impression/image/569e4a974c9d7b5f3b4005cb/locator_headquarter.jpeg"
  },
  "city": {
    "title": "Konstanz, Deutschland",
    "place_id": "ChIJWx8MOBv2mkcR0JnfpbdrHwQ",
    "id": "ChIJWx8MOBv2mkcR0JnfpbdrHwQ"
  },
  "public": true,
  "geotag": {
    "type": "Point",
    "coordinates": [
      9.169753789901733,
      47.66868204997508
    ]
  }
}
```

**Comment by SteffenGorenflo** [comment]
file upload and cor is ready. waiting for ms-location

**Comment by sigoetti** [comment]
this is so far what actually ends in the db:

``` json
{
  "user_id": "56aa58eb8fe80f8e306dbf1d",
  "title": "mein magic",
  "categories": [
    [
      "eine",
      "zwei"
    ]
  ],
  "favorites": [
    []
  ],
  "description": null,
  "images": {
    "xlarge": "/api/v2/locations/impression/image/56ad37d61387fbd4330cabf1/magicoutthere.jpg",
    "large": "/api/v2/locations/impression/image/56ad37d61387fbd4330cabf2/magicoutthere.jpg",
    "normal": "/api/v2/locations/impression/image/56ad37d61387fbd4330cabf3/magicoutthere.jpg",
    "small": "/api/v2/locations/impression/image/56ad37d61387fbd4330cabf4/magicoutthere.jpg"
  },
  "city": {
    "title": "here goes the city name",
    "place_id": "hIJWx8MOBv2mkcR0JnfpbdrHwQ",
    "id": "ChIJWx8MOBv2mkcR0JnfpbdrHwQ"
  },
  "public": true,
  "geotag": {
    "type": "Point",
    "coordinates": [
      9.775,
      47.6814
    ]
  },
  "create_date": "2016-01-30T22:23:18.845Z",
  "modified_date": "2016-01-30T22:23:18.845Z",
  "_id": "56ad37d67d9ab17102715fb8"
}
```

**Comment by timoweiss** [comment]
looks good to me, except:

``` javascript
"categories": [
    [
      "eine",
      "zwei"
    ]
  ],
  "favorites": [
    []
  ],
```

(there are arrays inside the arrays ;))

**Comment by sigoetti** [comment]
i will remove "id" in "city" as google states here: https://developers.google.com/places/web-service/search

> id contains a unique stable identifier denoting this place. This identifier may not be used to retrieve information about this place, but is guaranteed to be valid across sessions. It can be used to consolidate data about this place, and to verify the identity of a place across separate searches. Note: The id is now deprecated in favor of place_id. See the [deprecation notice](https://developers.google.com/places/web-service/search#deprecation) on this page.

**Comment by sigoetti** [comment]
now it's like that:

``` json

{
  "user_id": "56aa58eb8fe80f8e306dbf1d",
  "title": "magic 2",
  "categories": [
    "eine",
    "zwei"
  ],
  "favorites": [],
  "description": null,
  "images": {
    "xlarge": "/api/v2/locations/impression/image/56adf4731387fbd4330cabf9/magicoutthere.jpg",
    "large": "/api/v2/locations/impression/image/56adf4731387fbd4330cabfa/magicoutthere.jpg",
    "normal": "/api/v2/locations/impression/image/56adf4731387fbd4330cabfb/magicoutthere.jpg",
    "small": "/api/v2/locations/impression/image/56adf4731387fbd4330cabfc/magicoutthere.jpg"
  },
  "city": {
    "title": "here goes the city name",
    "place_id": "hIJWx8MOBv2mkcR0JnfpbdrHwQ"
  },
  "public": true,
  "geotag": {
    "type": "Point",
    "coordinates": [
      9.775,
      47.6814
    ]
  },
  "create_date": "2016-01-31T11:48:04.624Z",
  "modified_date": "2016-01-31T11:48:04.624Z",
  "_id": "56adf474f1911cde29f99e98"
}

```

**Comment by sigoetti** [comment]
was merged into cor in #58
